### ISSUE TITLE
sauceclient call to post test result is error prone

### ISSUE THREAD

**Issue by pgagne** [SatelliteQE/airgun#418]
https://github.com/SatelliteQE/airgun/blob/f4bc16c370545968db48835cc452829e3d317e47/airgun/browser.py#L321-L341

From my understanding from researching this is only so that the test result in Saucelabs it self will be shown as Pass/Fail. Without it, it would be just shown as "Complete" in saucelabs.

https://wiki.saucelabs.com/display/DOCS/Setting+Test+Status+to+Pass+or+Fail

On my local setup I have had many issues with this (SSL, and some others). Also, if you forget to install `sauceclient` (in `requirements-optional.txt`) you get a error. This code seems to be the only user of the `sauceclient` package

I was going to try to investigate more and fix the issues, but I started wondering what, if any value this feature has in the first place. I have never needed it. Also, I am not sure we want to be in the business of trying to hit a rest API (that may or may not be available) everytime a test ends. If it does fail, pytest marks the test as failed. 

Does anyone mind if I remove this?

**Comment by mirekdlugosz** [comment]
I see usefulness of this, but not in our context. We have never used SauceLabs for test results tracking, only as test execution platform. We have Polarion and ReportPortal for extracting and investigating failures. Nobody was going to SauceLabs to see if their tests failed, only to see **how** they failed.

On top of that, we aren't using Sauce in Satellite since September. This is essentially dead code for us.

@ldjebran how does it look in Insights? Would you mind if code pointed out by Perry is removed?

**Comment by flatfender** [comment]
So my feeling is that we should have a way to call sauce labs instead of Zalenium, in case we want to do "other browser" testing at some point.  So I see two issues here. 1. Reporting we don't use and triggering on sauce labs reporting when we're using Zalenium. 2. The ability to use either.  I'd make the reporting work if we over-ride Zalenium with sauce labs. This should be in the setup of the test run.

**Comment by mirekdlugosz** [comment]
@flatfender just so we are on the same page:
1. We don't report anything to Sauce when running under Zalenium
2. We are not thinking about removing Sauce support. It stays. If we decide Zalenium doesn't work for us and we want to get back to Sauce, we will be able to.
3. We already have config switches to decide if test run will be executed on Zalenium or Sauce (or local browser). We can run one test session on Zalenium and another on Sauce. We will still have that ability after removing code in question.

This is only about telling Sauce if test (that ran on Sauce) is "passed" or "failed". When code is removed, we will lose ability to say "hey Sauce, give me list of all failed tests since $DATE". We have report portal exactly to answer questions like that.
Personally, I never went to Sauce to see which tests failed - I already knew what failed and I was working on fixing it. I only opened Sauce to discover why exactly it failed. And removal of that code will not affect this workflow.

**Comment by pgagne** [comment]
Thanks for the feed back everyone, i am going to move forward with a PR to remove this.

**Comment by rplevka** [comment]
@pgagne can you elaborate more on the failures?
I have never seen any issues when I used them on my setup. This looks like a harsh decision.

**Comment by pgagne** [comment]
Here is one:
```
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
venv/lib/python3.6/site-packages/airgun/session.py:213: in __exit__
    self._factory.finalize(passed)
venv/lib/python3.6/site-packages/airgun/browser.py:160: in finalize
    return self._finalize_saucelabs_browser(passed)
venv/lib/python3.6/site-packages/airgun/browser.py:341: in _finalize_saucelabs_browser
    client.jobs.update_job(self._webdriver.session_id, **kwargs)
venv/lib/python3.6/site-packages/sauceclient.py:291: in update_job
    return self.client.request(method, endpoint, body=body)
venv/lib/python3.6/site-packages/sauceclient.py:84: in request
    connection.request(method, url, body, headers=headers)
/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/http/client.py:1239: in request
    self._send_request(method, url, body, headers, encode_chunked)
/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/http/client.py:1285: in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/http/client.py:1234: in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/http/client.py:1026: in _send_output
    self.send(msg)
/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/http/client.py:964: in send
    self.connect()
/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/http/client.py:1400: in connect
    server_hostname=server_hostname)
/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/ssl.py:407: in wrap_socket
    _context=self, _session=session)
/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/ssl.py:814: in __init__
    self.do_handshake()
/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/ssl.py:1068: in do_handshake
    self._sslobj.do_handshake()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <ssl.SSLObject object at 0x1055784a8>

    def do_handshake(self):
        """Start the SSL/TLS handshake."""
>       self._sslobj.do_handshake()
E       ssl.SSLError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:833)

/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/ssl.py:689: SSLError

```

It's totally possible that this is some weird macOS python SSL specific thing (since this job runs fine on standalone-automation without removing the code). 

Regardless, I don't think that we should be calling a remote REST API we have no control over by default (ie without the user explicitly setting something in the config file, just having install requirements-optional.txt shouldn't be all thats needed). 

@rplevka If you think there is value, perhaps you can a config file option or something to enable this reporting, perhaps it can be part of enabling test result reporting for zallenium as well

**Comment by omaciel** [comment]
I agree with @rplevka that we shouldn't be too quick to make changes here. While I understand that we're moving (or have already done so) to Zelenium, I don't think that we want to remove this functionality just because it may be failing on MacOS environment.

**Comment by pgagne** [comment]
Alright, I will put this on hold for now. I will revisit this after QECampX rdu. Can someone mark this as DONOTMERGE

**Comment by rplevka** [comment]
ok,
so I got back to this and read some more docs and it seems there should be much more elegant way of reporting the status of the testcase back to saucelabs:
https://github.com/saucelabs-sample-test-frameworks/Python-Pytest-Selenium/blob/master/conftest.py#L90

- haven't tried this but if this works, I'm happy to use this instead of invoking a parralel HTTP connections to the saucelabs.
- also, I reconsidered all the pros and cons and I'd be fine with removal of the mentioned code also.
